stylegan_code_finder/train_with_lightning.py

```diff
@@ -0,0 +1,151 @@
+    with open(config['class_to_color_map']) as f:
+        class_to_color_map = json.load(f)
+    assert len(class_to_color_map) == config['num_classes'], f'The number of classes in the class_to_color_map must ' \
+                                                             f'be equal to the num_classes in the config'
+
+
+def setup_distributed(mpi_backend: str, rank: int, world_size: int):
+    os.environ['MASTER_ADDR'] = 'localhost'
+    os.environ['MASTER_PORT'] = '12355'
+    dist.init_process_group(mpi_backend, rank=rank, world_size=world_size)
+
+
+def cleanup():
+    dist.destroy_process_group()
+
+
+def main(rank: int, args: argparse.Namespace, world_size: int):
+    if world_size == 1 and os.environ.get('REMOTE_PYCHARM_DEBUG_SESSION', False):
+        import pydevd_pycharm
+
+        logging.info('Connecting to debugger...')
+        pydevd_pycharm.settrace('localhost', port=int(os.environ.get('REMOTE_PYCHARM_DEBUG_PORT')),
+                                stdoutToServer=True, stderrToServer=True, suspend=False)
+        logging.info('Done')
+    global_config.debug = args.debug
+
+    config = load_yaml_config(args.config)
+    config = merge_config_and_args(config, args)
+    sanity_check_config(config)
+
+    if world_size > 1:
+        setup_distributed(args.mpi_backend, rank, world_size)
+
+    train_data_loader = get_data_loader(Path(config['train_json']), config['dataset'], args, config,
+                                        original_generator_config_path=args.original_generator_config_path)
+    if args.validation_json is not None:
+        val_data_loader = get_data_loader(Path(config['validation_json']), config['dataset'], args, config,
+                                          validation=True,
+                                          original_generator_config_path=args.original_generator_config_path)
+    else:
+        val_data_loader = None
+
+    train_builder_class = get_train_builder_class(config)
+    training_builder = train_builder_class(config, train_data_loader, val_data_loader, rank=rank, world_size=world_size)
+
+    pathlib.Path(args.log_dir, args.log_name).mkdir(parents=True, exist_ok=True)
+
+    logging.info("Initializing wandb... ")
+    logger = WandbLogger(name=args.log_name, save_dir=args.log_dir, project=args.wandb_project_name)
+    logging.info("done")
+
+    plot_data_loader = val_data_loader
+    if val_data_loader is None:
+        plot_data_loader = train_data_loader
+    segmentation_plotter = SegmentationPlotter(plot_data_loader, config)
+    segmenter_class = get_segmenter_class(config)
+    segmenter = segmenter_class(training_builder, config, segmentation_plotter, logger)
+
+    #snapshotter = training_builder.get_snapshotter()
+    #if snapshotter is not None:
+    #    trainer.extend(snapshotter)
+
+    #image_plotter = training_builder.get_image_plotter()
+    #if image_plotter is not None:
+    #    trainer.extend(image_plotter)
```

Comment:
Remove but make sure that Lightning takes proper snapshots. (Use `ModelCheckpoint` callback and only saved best checkpoints to minimize needed storage)